modules/storage-page-memory/src/test/java/org/apache/ignite/internal/storage/pagememory/index/InlineUtilsTest.java

```diff
@@ -0,0 +1,290 @@
+
+        assertEquals(2, inlineSize(nativeType = NativeTypes.INT16));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(4, inlineSize(nativeType = NativeTypes.INT32));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(8, inlineSize(nativeType = NativeTypes.INT64));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(4, inlineSize(nativeType = NativeTypes.FLOAT));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(8, inlineSize(nativeType = NativeTypes.DOUBLE));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(16, inlineSize(nativeType = NativeTypes.UUID));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(1, inlineSize(nativeType = NativeTypes.bitmaskOf(8)));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(3, inlineSize(nativeType = NativeTypes.DATE));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(4, inlineSize(nativeType = NativeTypes.time()));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(9, inlineSize(nativeType = NativeTypes.datetime()));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(12, inlineSize(nativeType = NativeTypes.timestamp()));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        // Variable length type checking.
+
+        assertEquals(BIG_NUMBER_INLINE_SIZE, inlineSize(nativeType = NativeTypes.decimalOf(1, 1)));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(BIG_NUMBER_INLINE_SIZE, inlineSize(nativeType = NativeTypes.decimalOf(100, 1)));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(7, inlineSize(nativeType = NativeTypes.stringOf(7)));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(MAX_VARLEN_INLINE_SIZE, inlineSize(nativeType = NativeTypes.stringOf(Integer.MAX_VALUE)));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(9, inlineSize(nativeType = NativeTypes.blobOf(9)));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(MAX_VARLEN_INLINE_SIZE, inlineSize(nativeType = NativeTypes.blobOf(Integer.MAX_VALUE)));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(BIG_NUMBER_INLINE_SIZE, inlineSize(nativeType = NativeTypes.numberOf(1)));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(BIG_NUMBER_INLINE_SIZE, inlineSize(nativeType = NativeTypes.numberOf(100)));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        // Let's check that all types have been checked.
+        assertThat(nativeTypeSpecs, empty());
+    }
+
+    @Test
+    void testBinaryTupleInlineSize() {
+        IndexDescriptor indexDescriptor = testIndexDescriptor();
+
+        assertEquals(BinaryTupleCommon.HEADER_SIZE, binaryTupleInlineSize(indexDescriptor));
```

Comment:
This is funny, you know. I'd expect an exception, who would want to make a tuple with 0 columns.
On the other hand, can this be useful somewhere? Just asking

Reply:
I don't know yet, I can add an assertion